hype-submitter_2.11/src/main/scala/com/spotify/hype/package.scala

```diff
@@ -26,27 +27,52 @@ package object hype {
       model.ContainerEngineCluster.containerEngineCluster(project, zone, cluster)
   }
 
-  def withSubmitter(cluster: model.ContainerEngineCluster, stagingBucket: String)
-                   (fn: (Submitter) => Unit): Unit = {
-    val submitter = Submitter.create(stagingBucket, cluster)
-    try fn(submitter)
-    finally submitter.close()
+  object GkeSubmitter {
+    def apply(stagingLocation: String, cluster: model.ContainerEngineCluster): Submitter =
+      setupShutdown(Submitter.create(stagingLocation, cluster))
   }
 
-  def withLocalSubmitter(dockerCluster: model.DockerCluster = model.DockerCluster.dockerCluster())
-                        (fn: (Submitter) => Unit): Unit = {
-    val submitter = Submitter.createLocal(dockerCluster)
-    try fn(submitter)
-    finally submitter.close()
+  object LocalSubmitter {
+    def apply(keepContainer: Boolean = false,
+              keepTerminationLog: Boolean = false,
+              keepVolumes: Boolean = false): Submitter =
+      setupShutdown(Submitter.createLocal(
+        dockerCluster(keepContainer, keepTerminationLog, keepVolumes)))
   }
 
-  type HypeFn[A] = util.Fn[A]
+  implicit class HFnOps[T](val hfn: HFn[T]) extends AnyVal {
+    def run(implicit submitter: Submitter, env: RunEnvironment): T = #!
 
-  implicit def toFn[A](fn: () => A): Fn[A] = new Fn[A] {
+    def #!(implicit submitter: Submitter, baseEnv: RunEnvironment): T =
```

Comment:
@rouzwawi Wondering how many people know of the operator `!`. Perhaps `#!` would be obscure to a lot of people. 
Might be worth it to add a more explicitly named operator (which also leverages implicits)?